### PR TITLE
Change Feed: Adds ChangeFeedIterator in GA

### LINKED ISSUE #2503: Code typo in cosmos getdatabase example
In first example here:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.cosmosclient.getdatabase?view=azure-dotnet

There is a typo:
("myDatabaseId"];

Should be:
("myDatabaseId");

replace bracket with parenthesis. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 440cc9dd-9e0b-44f8-f31a-36cdd97d5bbd
* Version Independent ID: 1ab2bf08-78df-a6c2-4c11-f021f8942a64
* Content: [CosmosClient.GetDatabase(String) Method (Microsoft.Azure.Cosmos) - Azure for .NET Developers](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.cosmosclient.getdatabase?view=azure-dotnet)
* Content Source: [xml/Microsoft.Azure.Cosmos/CosmosClient.xml](https://github.com/Azure/azure-docs-sdk-dotnet/blob/master/xml/Microsoft.Azure.Cosmos/CosmosClient.xml)
* Service: **cosmos-db**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

### PR DESCRIPTION
**Pull request by ealsur**
## Description

This PR promotes the ChangeFeedIterator APIs to GA and moves them out of the PREVIEW package.

The GAed APIs are:

* `Container.GetChangeFeedIterator`
* `Container.GetChangeFeedStreamIterator`
* `Container.GetFeedRangesAsync`
* `Container.GetPartitionKeyRangeAsync`
* `FeedRange`
* `ChangeFeedRequestOptions`

image

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Closing issues

Closes #2503

### CODE REVIEW

**Review by FabianMeiswinkel — APPROVED** [review]

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/DotNetSDKAPI.json [hunk 1]
```diff
@@ -405,6 +421,128 @@
...
+            "CompilerGeneratedAttribute"
+          ],
+          "MethodInfo": "Void set_EmitOldContinuationToken(Boolean);IsAbstract:False;IsStatic:False;IsVirtual:False;IsGenericMethod:False;IsConstructor:False;IsFinal:False;"
+        },
+        "Void set_IfMatchEtag(System.String)": {
+          "Type": "Method",
+          "Attributes": [],
+          "MethodInfo": "Void set_IfMatchEtag(System.String);IsAbstract:False;IsStatic:False;IsVirtual:False;IsGenericMethod:False;IsConstructor:False;IsFinal:False;"
+        },
+        "Void set_IfNoneMatchEtag(System.String)": {
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/DotNetSDKAPI.json:492]: Can you possibly add the API diff to the PR description like it is done for the releases? That way it's clear list of all the new APIs being exposed.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/DotNetSDKAPI.json [hunk 2]
```diff
@@ -405,6 +421,128 @@
       },
       "NestedTypes": {}
     },
+    "Microsoft.Azure.Cosmos.ChangeFeedRequestOptions;Microsoft.Azure.Cosmos.RequestOptions;IsAbstract:False;IsSealed:True;IsInterface:False;IsEnum:False;IsClass:True;IsValueType:False;IsNested:False;IsGenericType:False;IsSerializable:False": {
+      "Subclasses": {},
+      "Members": {
+        "Boolean EmitOldContinuationToken": {
+          "Type": "Property",
+          "Attributes": [],
+          "MethodInfo": "Boolean EmitOldContinuationToken;CanRead:True;CanWrite:True;Boolean get_EmitOldContinuationToken();IsAbstract:False;IsStatic:False;IsVirtual:False;IsGenericMethod:False;IsConstructor:False;IsFinal:False;Void set_EmitOldContinuationToken(Boolean);IsAbstract:False;IsStatic:False;IsVirtual:False;IsGenericMethod:False;IsConstructor:False;IsFinal:False;"
```
[the remark this answers is not in the document]
Comment (ealsur) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/DotNetSDKAPI.json:430]: It was part of the options, I truly don't know the use of it.

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 3]
```diff
@@ -1363,6 +1344,25 @@ public abstract class Container
             FeedRange feedRange,
             CancellationToken cancellationToken = default);
 
+#if INTERNAL
```
[the remark this answers is not in the document]
Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:1347]: Well, it's part of the preview APIs that use FeedRange, guess it's not critical but someone might have been using it. I can mantain it as preview for now

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/DotNetSDKAPI.json (hunk 1 above)
Comment (ealsur) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/DotNetSDKAPI.json:473]: Done

File: Microsoft.Azure.Cosmos/src/FeedRange/FeedRange.cs [hunk 4]
```diff
@@ -10,12 +10,7 @@ namespace Microsoft.Azure.Cosmos
     /// Represents a unit of feed consumption that can be used as unit of parallelism.
     /// </summary>
     [Serializable]
-#if PREVIEW || INTERNAL
-    public
-#else
-    internal
-#endif
-    abstract class FeedRange
+    public abstract class FeedRange
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/FeedRange/FeedRange.cs:13]: PartitionKey.None requires the container to be able to correctly determine what the json string value is. How is this working correctly?

```c#
FeedRange feedRange = FeedRange.FromPartitionKey(PartitionKey.None);
string range = feedRange.ToJsonString();
```

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/FeedRange/FeedRange.cs:13]: This is a good catch that was not considered during the initial implementation, fixed it

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**